### PR TITLE
Add Node.js workflow

### PR DESCRIPTION
**Pull request by KSXGitHub**
Resolve https://github.com/iter-tools/iter-tools/issues/273

### CODE REVIEW

**Comment by KSXGitHub** [comment]
This is what I meant by shorter feedback loop:

checks screenshot

We will see every individual check at once.

**Comment by conartist6** [comment]
Let me know when this is ready to go

**Comment by KSXGitHub** [comment]
> Let me know when this is ready to go

Yes.

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 10.x
+        npm-script:
+          - test:js
+          - test:types
+          - verify
```
Comment (conartist6) [inline .github/workflows/node.yml]: Don't run `verify`. Verify just runs all these other steps that you're calling individually, so there's no point in doing that, particularly if Travis is also running the complete verify script.

File: .github/workflows/node.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 10.x
+        npm-script:
```
Comment (conartist6) [inline .github/workflows/node.yml]: This appears to be an abuse of the actions API. Matrix is intended to allow you to choose node and OS versions. If you want to run different kinds of tests, you should be defining multiple jobs.

Why do you need to define multiple jobs? ALL the jobs which you have listed here require the code generator to be run first. Only the jobs API supports the setup you need, allowing you to set up a sequential step, after which several different kinds of tasks are executed in parallel. That API is documented [here](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idneeds)

Comment (KSXGitHub) [reply .github/workflows/node.yml]: Very well, I will do as you wish when I'm back at my computer.

Comment (KSXGitHub) [reply .github/workflows/node.yml]: > ALL the jobs which you have listed here require the code generator to be run first. Only the jobs API supports the setup you need, allowing you to set up a sequential step, after which several different kinds of tasks are executed in parallel.

I just discovered that is not how it works ([I tried on a different repo](https://github.com/ksxnodemodules/ts-pipe-compose/runs/228072115)): Each job run on their own virtual machine, completely isolated from each other, this is evidence by the fact that `runs_on` is required for every job. That means that if a "code generator" job already generate the code, other jobs won't have access to the generated code even if it depends on "code generator".

Comment (KSXGitHub) [reply .github/workflows/node.yml]: So, do you still want to divide the matrix into multiple jobs?

**Comment by KSXGitHub** [comment]
@conartist6 It is done.

We barely utilize GitHub Actions for faster testing, not its full potential: Such as creating a bot to automatically comment upon test completion, review a pull request, build documentation website and publish it, build package and publish it, etc.

As for https://github.com/iter-tools/iter-tools/issues/293#issuecomment-533199411, I can't because you haven't created it yet.

**Comment by conartist6** [comment]
I have now switched to actions. I didn't do anything quite this complex. Currently I'm only testing on a single node version (v12), but I may yet decide to create additional jobs (or workflows) to test on older node versions.

**Pull request closed without merging**